packages/@aws-cdk/aws-scheduler-alpha/lib/group.ts

```diff
@@ -0,0 +1,372 @@
+   *
+   * @see https://docs.aws.amazon.com/scheduler/latest/UserGuide/scheduler-quotas.html
+   *
+   * @default - sum over 5 minutes
+   */
+  metricThrottled(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * Metric for all invocation attempts.
+   *
+   * @default - sum over 5 minutes
+   */
+  metricAttempts(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * Emitted when the target returns an exception after EventBridge Scheduler calls the target API.
+   *
+   * @default - sum over 5 minutes
+   */
+  metricTargetErrors(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * Metric for invocation failures due to API throttling by the target.
+   *
+   * @default - sum over 5 minutes
+   */
+  metricTargetThrottled(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * Metric for dropped invocations when EventBridge Scheduler stops attempting to invoke the target after a schedule's retry policy has been exhausted.
+   *
+   * @default - sum over 5 minutes
+   */
+  metricDropped(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * Metric for invocations delivered to the DLQ
+   *
+   * @default - sum over 5 minutes
+   */
+  metricSentToDLQ(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * Metric for failed invocations that also failed to deliver to DLQ.
+   *
+   * @default - sum over 5 minutes
+   */
+  metricFailedToBeSentToDLQ(errorCode?: string, props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * Metric for delivery of failed invocations to DLQ when the payload of the event sent to the DLQ exceeds the maximum size allowed by Amazon SQS.
+   *
+   * @default - sum over 5 minutes
+   */
+  metricSentToDLQTrunacted(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * Grant the indicated permissions on this group to the given principal
+   */
+  grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant;
+  /**
+   * Grant list and get schedule permissions for schedules in this group to the given principal
+   */
+  grantReadSchedules(identity: iam.IGrantable): iam.Grant;
+  /**
+   * Grant create and update schedule permissions for schedules in this group to the given principal
+   */
+  grantWriteSchedules(identity: iam.IGrantable): iam.Grant;
+  /**
+   * Grant delete schedule permission for schedules in this group to the given principal
+   */
+  grantDeleteSchedules(identity: iam.IGrantable): iam.Grant
+}
+
+abstract class GroupBase extends Resource implements IGroup {
+  /**
+   * The name of the schedule group
+   *
+   * @attribute
+   */
+  abstract readonly groupName: string;
+
+  /**
+   * The arn of the schedule group
+   *
+   * @attribute
+   */
+  abstract readonly groupArn: string;
+
+  addSchedules(...schedules: ISchedule[]): void {
+    schedules.forEach(schedule => {
+      // Temporary using ISchedule as parameter for this method,
```

Comment:
info: why temporarily using ISchedule?

Reply:
Because `Schedule` is private now, but `ISchedule` is not. If I use private class `Schedule` as type for the parameter of a public method - the build will fail

Reply:
I mean that we _should_ be using `ISchedule` here

Reply:
> I mean that we should be using ISchedule here

Hm, good point. Then we would need to make a mutable property `group` on interface `ISchedule`.  I originally thought that we could import schedules from ARN in the future (e.g. from another stack / account), so that's why it's better to explicitly point to `Schedule` as a class.

Reply:
Importing a schedule from ARN should return an `ISchedule` class. We can't mutate imports so they cannot be created as a `Schedule`.